### PR TITLE
fix(pyspark): unwind catalog/database settings in same order they were set

### PR DESCRIPTION
**Pull request by gforsyth**
We set the catalog using a context manager, and we set the database also
using a context manager. There is a weird edge case:

set catalog to comms_media_dev (succeeds)
set database to dart_extensions within comms_media_dev (succeeds)

then we write the table, great! Now we try to change catalog and database back in reverse order and...

set database to default (the previous value that we saved) within comms_media_dev (fails, you do not have permission to access that)

set catalog back to spark_catalog (or previous value, but we never get here because of the previous error.

So what we need to do is instead:

set catalog
set database
write table
set catalog back
set database back

It would be really great if spark would allow for setting both of these values at the same time, but that is apparently not a thing.

xref #9038

### CODE REVIEW

**Comment by gforsyth** [comment]
I forgot to include `catalog` in the arguments to `self.table` so we were looking for a table in a different DB that we _hadn't_ just created.

File: ibis/backends/pyspark/tests/test_client.py [hunk 1]
```diff
@@ -20,3 +23,19 @@ def test_catalog_db_args(con, monkeypatch):
...
+def test_create_table_no_catalog(con, monkeypatch):
+    monkeypatch.setattr(ibis.options, "default_backend", con)
+    t = ibis.memtable({"epoch": [1712848119, 1712848121, 1712848155]})
+
+    # create a table in specified catalog and db
+    con.create_table("t2", database=("default"), obj=t, overwrite=True)
+
+    assert "t2" not in con.list_tables()
+    assert "t2" in con.list_tables(database="default")
+    assert "t2" in con.list_tables(database=("default"))
```
Comment (cpcloud) [inline ibis/backends/pyspark/tests/test_client.py:37]: Did you mean to write something else here? `"default"` and `("default")` are equivalent Python expressions.

Comment (gforsyth) [reply ibis/backends/pyspark/tests/test_client.py:37]: oh, that was just unthinking editing -- I'll rip that out

**Review by cpcloud — APPROVED** [review]

**Comment by markdruffel-8451** [comment]
@gforsyth Just a heads up this is now working for my use case (multiple catalogs). Thanks so much for all your help and definitely feel free to hit me up if you ever need help testing pyspark backend 👍

**Pull request merged**